### ISSUE TITLE
Split changesets to prevent huge areas

### ISSUE THREAD

**Issue by Norwi** [westnordost/StreetComplete#2053]
**Use case**
When you are travelling you don't necessarily want to upload the changes immediately (e.g. you don't want to 'ping' your location every half an hour). But if you upload the changes later it could become a huge changeset (in terms of area).
For example: [openstreetmap.org/changeset/89030893](https://www.openstreetmap.org/changeset/89030893). It has only a few changes but takes up well over 10 000km².

**Proposed Solution**
Maybe it could be implemented by checking the distance between nodes and if they are more than x km apart, they will be split into two changesets. Another way would be to let the user draw a line and let them determine what should be part of the changeset.

**Comment by westnordost** [comment]
I'd rather not automatically split it. The worst that will happen is that the changeset is a little difficult to review.

I see it as the responsibility of the contributor to not collect too many changes before he uploads it. No other editor splits changesets automatically or offers an interface to select (git-like) only certain changes to include.

**Comment by JesseWeinstein** [comment]
Just leaving this other example of a large changeset here: https://www.openstreetmap.org/changeset/90817239

Your reasons for marking it as wontfix make sense, but it seems good to track how often the problem happens, as that might be relevant.

In any case, thank you VERY much for StreetComplete; I've really enjoyed contributing using it.

**Comment by lgommans** [comment]
@westnordost 

> I see it as the responsibility of the contributor to not collect too many changes before he uploads it.

This would be fine if it worked, but StreetComplete groups changes together regardless of upload status. I was on WiFi, so changes were uploaded as I made them (i.e. before making the next change, the previous one was already uploaded), yet I still got a huge changeset.

I got a complaint today on changeset [95363018](https://www.openstreetmap.org/changeset/95363018) for being too large geographically. I agree it's up to the contributor, me, to use the editor in a way that works for everyone in the community, but I don't know how to do that and uploading regularly isn't the solution, apparently.

How can I make sure it starts a new changeset? Should I close and reopen the app when switching areas, or is there some other trick? (It seems to split on quest type, but I may not always know the answer to other quests in the area than the type I'm solving.)

**Comment by westnordost** [comment]
Quests of the same type are sorted into the same changeset if they were solved (that is: uploaded) less than 20min apart IIRC.

**Comment by matkoniecz** [comment]
> No other editor splits changesets automatically

Though other editors have user manually selecting moment when new changeset will be created.

Still, SC is supposed to be used for on-the-ground survey. If someone maps remotely - or even, worse based on memory and guessing then it is not SC that such misuse will end in substandard results.

**Comment by lgommans** [comment]
@matkoniecz I get that you disagree here; we talked about it in chat and we clearly differ in opinion on whether solving specific (easy) quests for areas where you're very familiar is a reasonable thing to do. You can opine that it is misuse, but telling a person not to scratch their itch doesn't work as well as providing ways of making it not itch. I think it's useful (unless we find data to show otherwise; feel free to tag >1km distant changes with distant=yes or something and we can have a look how people are using it) and I'd simply like to make the changesets manageable for others in the community.

Anyway, in conclusion: Solutions:
- Making sure you do a different quest type as last or first change before or after switching area
- Making sure 20 minutes passed since the last upload before uploading changes in a new area

With auto-uploading on WiFi disabled (suggested in chat by Mateusz), those sound like workable solutions to me :)

**Comment by riQQ** [comment]
Search keywords: big / large bounding box
